### PR TITLE
Feat/1205123214195396 coefficient

### PR DESCRIPTION
**Pull request by sfffaaa**
1. Add the trait for the delegator/collator reward calculator
2. Add a new mechanism for parachain staking, coefficient
3. Extract the origin reward calculator to the new pallet
4. Add migration to remove the original reward rate in the parachain staking pallet.

Note:
I haven't patched it to the claim mechanism branch, I will do that later.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205123214195396

### CODE REVIEW

File: runtime/peaq/src/lib.rs [hunk 1]
```diff
@@ -994,6 +1004,8 @@ construct_runtime!(
 		ParachainSystem: cumulus_pallet_parachain_system::{Pallet, Call, Storage, Inherent, Event<T>} = 24,
 		ParachainInfo: parachain_info::{Pallet, Storage, Config} = 25,
 		BlockReward: pallet_block_reward::{Pallet, Call, Storage, Config<T>, Event<T>} = 26,
+		// StakingFixedRewardCalculator: staking_fixed_percentage_reward::{Pallet, Call, Storage, Config, Event<T>} = 27,
```
Comment (neutrinoks) [inline runtime/peaq/src/lib.rs]: Why this place holder? Will it be integrated in the future? Or is it an alternative to `StakingCoefficientRewardCalculator`?

File: runtime/krest/src/lib.rs [hunk 2]
```diff
@@ -774,6 +774,9 @@ pub mod staking {
 	pub fn reward_rate_config() -> RewardRateInfo {
 		RewardRateInfo::new(Perquintill::from_percent(30), Perquintill::from_percent(70))
 	}
+	pub fn coefficient() -> u8 {
+		8
```
Comment (neutrinoks) [inline runtime/krest/src/lib.rs:778]: Why 8? :upside_down_face:

**Review by neutrinoks — APPROVED** [review]

File: runtime/krest/src/lib.rs (hunk 2 above)
Comment (sfffaaa) [reply runtime/krest/src/lib.rs:778]: 8 is a magic number from Ilya. 

> I changed the number to 8, as this way we could ensure that collators are almost always well compensated

File: runtime/peaq/src/lib.rs (hunk 1 above)
Comment (sfffaaa) [reply runtime/peaq/src/lib.rs]: Yep, you are right. It's an alternative to `StakingCoefficientRewardCalculator`. However, let me remove that.

**Pull request merged**